AutomatticTracks/src/main/java/com/automattic/android/tracks/DeviceInformation.java

```diff
@@ -113,10 +117,33 @@ public DeviceInformation(Context context) {
 
         mHasNFC = foundNFC;
         mHasTelephony = foundTelephony;
-        mDisplayMetrics = new DisplayMetrics();
 
+        mDisplayMetrics = new DisplayMetrics();
         Display display = ((WindowManager) mContext.getSystemService(Context.WINDOW_SERVICE)).getDefaultDisplay();
-        display.getMetrics(mDisplayMetrics);
+        if (display != null) {
+            display.getMetrics(mDisplayMetrics);
+            // since SDK_INT = 1; This doesn't include window decorations
+            mWidthPixels = mDisplayMetrics.widthPixels;
+            mHeightPixels = mDisplayMetrics.heightPixels;
+            // Try to load the real screen size now - This does include window decorations (statusbar bar/menu bar)
+            if (Build.VERSION.SDK_INT >= 14 && Build.VERSION.SDK_INT < 17) {
+                try {
+                    mWidthPixels = (int) Display.class.getMethod("getRawWidth").invoke(display);
+                    mHeightPixels = (int) Display.class.getMethod("getRawHeight").invoke(display);
+                } catch (Exception ignored) {
+                    Log.w(LOGTAG, "Unable to call getRawWidth/getRawHeight: " + ignored.getMessage());
+                }
+            } else if (Build.VERSION.SDK_INT >= 17) {
+                try {
+                    Point realSize = new Point();
+                    Display.class.getMethod("getRealSize", Point.class).invoke(display, realSize);
```

Comment:
Wondering, why does this need reflection? Can't we just call `getRealSize` directly here that it's guarded by the `if (Build.VERSION.SDK_INT >= 17)`?

Reply:
Once you start use reflection you can't do without it :)
I've removed the unnecessary reflection calls.